app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Blukon.java

```diff
@@ -426,29 +424,42 @@ public static void unBondIfBlukonAtInit() {
          */
         } else if (currentCommand.startsWith(GET_SENSOR_TIME_COMMAND) /*getSensorAge*/ && strRecCmd.startsWith(SINGLE_BLOCK_INFO_RESPONSE_PREFIX)) {
             cmdFound = 1;
-            Log.i(TAG, "SensorAge received");
 
             int sensorAge = sensorAge(buffer);
+            Log.d(TAG, "SensorAge received=" + sensorAge);
 
-            if (Pref.getBooleanDefaultFalse("external_blukon_algorithm") || getHistoricReadings) {
-                // Send the command to getHistoricData (read all blcoks from 0 to 0x2b)
-                Log.i(TAG, "getHistoricData (3)");
-                currentCommand = GET_HISTORIC_DATA_COMMAND_ALL_BLOCKS;
-                m_blockNumber = 0;
+            int currentSensorAge = Pref.getInt("nfc_sensor_age", 0);
+            Log.d(TAG, "current SensorAge=" + currentSensorAge);
+
+            //This is a new sensor, force read from serial
+            if (sensorAge < currentSensorAge) {
+                Log.i(TAG, "new sensor?");
+                JoH.clearRatelimit(BLUKON_DECODE_SERIAL_TIMER);// set to current time to force timer to be set back
+            }
+
+            if ((sensorAge >= 0) && (sensorAge < 200000)) {
+                Pref.setInt("nfc_sensor_age", sensorAge);//in min
+                //when getting historic, we use LibreAlarm Code and sensor age is not exactly same as calculated here
+                //to avoid warning, simply overide this flag
+                Pref.setBoolean("nfc_age_problem", false);
             } else {
-                /* LibreAlarmReceiver.CalculateFromDataTransferObject, called when processing historical data,
-                 * expects the sensor age not to be updated yet, so only update the sensor age when not retrieving history.
-                 */
-                if ((sensorAge >= 0) && (sensorAge < 200000)) {
-                    Pref.setInt("nfc_sensor_age", sensorAge);//in min
-                } else {
-                    Log.e(TAG, "Do not set 'nfc_sensor_age'");
-                }
-                currentCommand = GET_NOW_DATA_INDEX_COMMAND;
-                m_getNowGlucoseDataIndexCommand = true;//to avoid issue when gotNowDataIndex cmd could be same as getNowGlucoseData (case block=3)
-                Log.i(TAG, "getNowGlucoseDataIndexCommand");
+                Log.e(TAG, "Do not set 'nfc_sensor_age'");
             }
 
+            currentSensorAge = Pref.getInt("nfc_sensor_age", 0);
```

Comment:
I'm puzzled by the duplicate setting of currentSensorAge and currentCommand? If required, can you document?

Reply:
good catch, thanks!